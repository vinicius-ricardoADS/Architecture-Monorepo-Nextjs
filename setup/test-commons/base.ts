import type { Config } from "@jest/types";
import path from "path";

const config: Config.InitialOptions = {
  testEnvironment: 'jsdom',
  verbose: true,
  transform: {
    '\\.[jt]sx?$': ['babel-jest', {
      configFile: path.resolve(__dirname, '.babelrc'),
    }],
  }
}

export default config;
