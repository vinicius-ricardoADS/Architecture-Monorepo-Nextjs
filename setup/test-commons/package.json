{
  "name": "@architecture/test-commons",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "files": [
    "base.ts"
  ],
  "devDependencies": {
    "@testing-library/react": "15.0.7",
    "jest": "27.5.0",
    "ts-node": "10.9.2",
    "babel-jest": "28.0.3",
    "@babel/core": "7.24.6",
    "@babel/preset-env": "7.24.6",
    "@babel/preset-react": "7.24.6",
    "@babel/preset-typescript": "7.24.6"
  }
}
