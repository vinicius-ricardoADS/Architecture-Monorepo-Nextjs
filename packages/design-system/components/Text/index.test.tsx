import { Text } from ".";
import { customRender } from '@architecture/test-commons/react-testing-library';

const render = customRender();

describe('<Text />', () => {
  it('renders h1 tag', () => {
    const { 
      container 
    } = render(<Text tag="h1">Sample text</Text>);
    
    expect(container).toMatchSnapshot();
  })
})
