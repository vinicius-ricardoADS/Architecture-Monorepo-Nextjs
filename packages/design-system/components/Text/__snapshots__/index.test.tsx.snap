// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders h1 tag 1`] = `
<div>
  <h1
    class="sc-aYaIB jNjAcf"
  >
    Sample text
  </h1>
</div>
`;
